src/repository.js

```diff
@@ -187,6 +187,12 @@ module.exports = class Repository {
       .returning(fields)
       .insert(payload)
 
+    //.returning() is not supported by sqlite3 and will not have any effect, so we have to get last inserted row
+    if (this.runner().client && this.runner().client.driverName && (this.runner().client.driverName.includes('sqlite3'))) {
+      const lastInsertedID = await this.knex.raw(`SELECT last_insert_rowid();`)
+      return await this.findByID(lastInsertedID[0]["last_insert_rowid()"])
```

Comment:
`this.findByID` expect a id(s) from the entity. A "domain" ID.

`last_insert_rowid` return the PK only if (1) it exists a PK, (2) if it is a integer [1] and (3) if it is not composed PK. Otherwise it returns a number not related to a "domain" ID.

So, the current implementation seems very error-prone. Is this the only method to return the entity? 

Suggestion: 
If the insertion does not return a error, use the known IDs of the inserted record to retrieve the record.


[1] from [doc](https://www.sqlite.org/c3ref/last_insert_rowid.html): If the table has a column of type [INTEGER PRIMARY KEY](https://www.sqlite.org/lang_createtable.html#rowid) then that column is another alias for the rowid.

Reply:
I'm aware (also working on it) of this possible bug and will fix it ASAP.